coalib/testing/LocalBearTestHelper.py

```diff
@@ -52,6 +54,22 @@ class LocalBearTestHelper(unittest.TestCase):
     If you miss some methods, get in contact with us, we'll be happy to help!
     """
 
+    def assertComparableObjectsEqual(self, observed_result, expected_result):
+        self.assertEqual(len(observed_result), len(expected_result),
+                         msg='The number of elements in the observed result '
+                             'and expected result differ.')
+        for observed, expected in zip_longest(
+                observed_result, expected_result):
+            if (isinstance(observed, Comparable)
+                and isinstance(expected, Comparable)) and (
+                        type(observed) is type(expected)):
+                for attribute in type(observed).__compare_fields__:
+                    self.assertEqual(getattr(observed, attribute),
+                                     getattr(expected, attribute),
+                                     msg='{} mismatch.'.format(attribute))
```

Comment:
> Also the order of the attributes compared needs to be considered if the first difference is going to be the only mismatch shown.

Makes sense to display all flaws at once. :+1:

However, the ordering is actually not important imo.